include/deal.II/base/mpi_remote_point_evaluation.h

```diff
@@ -296,6 +296,126 @@ namespace Utilities
       std::vector<unsigned int> send_ptrs;
     };
 
+    namespace internal
+    {
+      /**
+       * Pack @p data and send it via MPI_Isend.
+       */
+      template <typename T>
+      std::enable_if_t<Utilities::MPI::is_mpi_type<T> == false, void>
+      pack_and_isend(T *                             data,
+                     const unsigned int              size,
+                     const unsigned int              rank,
+                     const unsigned int              tag,
+                     const MPI_Comm                  comm,
+                     std::vector<std::vector<char>> &buffers,
+                     std::vector<MPI_Request> &      requests)
+      {
+        requests.emplace_back(MPI_Request());
+
+        const int ierr = MPI_Isend(data,
+                                   size,
+                                   Utilities::MPI::mpi_type_id_for_type<T>,
+                                   rank,
+                                   tag,
+                                   comm,
+                                   &requests.back());
+        AssertThrowMPI(ierr);
+      }
+
+
+
+      /**
+       * Above function specialized for data types supported by MPI
+       * so that one can skip packing.
+       */
+      template <typename T>
+      std::enable_if_t<Utilities::MPI::is_mpi_type<T> == true, void>
+      pack_and_isend(T *                             data,
+                     const unsigned int              size,
+                     const unsigned int              rank,
+                     const unsigned int              tag,
+                     const MPI_Comm                  comm,
+                     std::vector<std::vector<char>> &buffers,
+                     std::vector<MPI_Request> &      requests)
+      {
+        requests.emplace_back(MPI_Request());
+
+        buffers.emplace_back(
+          Utilities::pack(std::vector<T>(data, data + size), false));
```

Comment:
You have this the wrong way around: This is the function with `Utilities::MPI::is_mpi_type<T> == true` for which you specifically don't have to pack, whereas for the function above where you don't pack you actually do have to pack.